### ISSUE TITLE
Document the size limitation when using compression under Python 2

### ISSUE THREAD

**Issue by ogrisel** [joblib/joblib#300]
As revealed in the discussion of #122, the Python 2.7 zlib module cannot compress very large arrays (maximum  is 2 ** 31 - 1 bytes) while Python 3 does not have this limitation.

We could even detect that  problem is the code before trying to compress and raise a more informative exception message instead.

**Comment by kjmph** [comment]
Note, upstream bug is here: https://bugs.python.org/issue23306

**Comment by lesteve** [comment]
Since #260 was merged, we are compressing by small chunks so this is no longer an issue, closing.

For completeness, here is a snippet that fails with joblib 0.9.4 and Python 2.7 but works fine with joblib master and Python 2.7:

```py
import numpy as np

import joblib

arr = np.zeros(2**31, dtype='uint8')
joblib.dump(arr, '/tmp/test.pkl', compress=True)
```

**Comment by ogrisel** [comment]
This is still an issue for `joblib.Parallel` does not no rely on `joblib.dump` because it need to do automated memmaping only if there is an array that is too large.

**Comment by ogrisel** [comment]
Actually I was wrong in re-opening this. I reacted to https://github.com/joblib/joblib/issues/122#issuecomment-240391849 which is actually not related to the original problem reported in #122: the problem that happens in `joblib.dump` with compression enabled initially reported in #122 is really fixed in joblib 0.10.

The other problem reported by https://github.com/joblib/joblib/issues/122#issuecomment-240391849 should be worked-around by letting `joblib.Parallel` do automated memmaping of very large numpy arrays.
